api/v1alpha3/virtualmachine_types.go

```diff
@@ -236,8 +236,44 @@ type VirtualMachineImageRef struct {
 	Name string `json:"name"`
 }
 
+// VirtualMachineCdromSpec describes the desired state of a CD-ROM device.
+type VirtualMachineCdromSpec struct {
+	// +required
+
+	// ImageName is the Kubernetes object name of the VirtualMachineImage or
+	// ClusterVirtualMachineImage resource of type ISO that is used as the
+	// backing for the CD-ROM device.
+	ImageName string `json:"imageName"`
+
+	// +optional
+	// +kubebuilder:validation:Enum=VirtualMachineImage;ClusterVirtualMachineImage
+	// +kubebuilder:default=VirtualMachineImage
+
+	// ImageKind describes the type of image used as the backing for the CD-ROM
+	// device. It may be either a VirtualMachineImage (default value) or a
+	// ClusterVirtualMachineImage.
+	ImageKind string `json:"imageKind,omitempty"`
+
+	// +optional
+
+	// Disconnect indicates whether the CD-ROM device should be disconnected
+	// from the VM. If omitted or set to false, the CD-ROM device is added and
+	// connected to the VM when the VM is powered on.
+	Disconnect bool `json:"connect,omitempty"`
```

Comment:
I assume the intent is that setting this to true also means the customer can expect the CD-ROM to be immediately disconnected for a running VM?